src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.Extensions.cs

```diff
@@ -81,7 +81,7 @@ private static int Shutdown()
         [SupportedOSPlatform("windows")]
         private static void InitializeApartmentState(ApartmentState state)
         {
-            Thread.CurrentThread.SetApartmentState(state);
+            Thread.CurrentThread.SetApartmentStateUnchecked(state, throwOnError: true);
```

Comment:
Is this change actually needed?

Reply:
Yeah, good catch, it isn't. I probably got lost in the dependency graph and thought we validate this with stringification too.